source/AI.cpp

```diff
@@ -2764,24 +2764,23 @@ void AI::DoSurveillance(Ship &ship, Command &command, shared_ptr<Ship> &target)
 		}
 
 		unsigned total = targetShips.size() + targetPlanets.size() + targetSystems.size();
+		// If there is nothing for this ship to scan, have it patrol the entire system
+		// instead of drifting or stopping.
 		if(!total)
-		{
-			// If there is nothing for this ship to scan, have it hold still
-			// instead of drifting away from the system center.
-			Stop(ship, command);
-			return;
-		}
-
-		unsigned index = Random::Int(total);
-		if(index < targetShips.size())
-			ship.SetTargetShip(targetShips[index]->shared_from_this());
+			DoPatrol(ship, command);
```

Comment:
Why not just `return;`? This path has nothing else to do in this method.
```suggestion
		{
			DoPatrol(ship, command);
			return;
		}
```